libs/designer-ui/src/lib/html/plugins/toolbar/helper/HTMLChangePlugin.tsx

```diff
@@ -1,35 +1,77 @@
 import type { ValueSegment } from '../../../../editor';
 import { convertStringToSegments } from '../../../../editor/base/utils/editorToSegment';
 import { getChildrenNodes } from '../../../../editor/base/utils/helper';
-import { cleanHtmlString, decodeSegmentValue, encodeSegmentValue } from './util';
+import {
+  decodeStringSegmentTokensInDomContext,
+  decodeStringSegmentTokensInLexicalContext,
+  encodeStringSegmentTokensInLexicalContext,
+} from '../../../../editor/base/utils/parsesegments';
+import {
+  cleanHtmlString,
+  cleanStyleAttribute,
+  encodeSegmentValueInLexicalContext,
+  getDomFromHtmlEditorString,
+  isAttributeSupportedByLexical,
+  isHtmlStringValueSafeForLexical,
+} from './util';
 import { $generateHtmlFromNodes } from '@lexical/html';
 import { OnChangePlugin } from '@lexical/react/LexicalOnChangePlugin';
 import type { EditorState, LexicalEditor } from 'lexical';
 import { $getRoot } from 'lexical';
 
 interface HTMLChangePluginProps {
+  isValuePlaintext: boolean;
+  loadParameterValueFromString?: (value: string) => ValueSegment[];
+  setIsValuePlaintext: (isValuePlaintext: boolean) => void;
   setValue: (newVal: ValueSegment[]) => void;
 }
 
-export const HTMLChangePlugin = ({ setValue }: HTMLChangePluginProps) => {
+export const HTMLChangePlugin = ({
+  isValuePlaintext,
+  loadParameterValueFromString,
+  setIsValuePlaintext,
+  setValue,
+}: HTMLChangePluginProps) => {
   const onChange = (editorState: EditorState, editor: LexicalEditor) => {
```

Comment:
I've found one bug where if you add a token outside of an element in the raw editor, then switch, it does something janky:

![html-editor-bug-1](https://github.com/Azure/LogicAppsUX/assets/1350074/a3bcfcc5-d7a9-49f7-8c08-02d6bdcf30f8)

Reply:
this might be something we want to look at in a future PR, but if im gonna guess its when converting it from lexical editor to dom, its inferring the html elements to get that

Reply:
Yeah, I think everything at the top level needs to be a known tag type, but will leave this as an open bug to address.